### DISCUSSION TITLE
Comentarios de lo que me voy encontrando durante la revisión

### DISCUSSION THREAD

**Discussion by mmasias** [mmasias/24-25-PRG1#584] [General]
Para que no se me pasen, los voy poniendo aquí.

Por supuesto, abierto a comentarios!!!

**Comment by mmasias** [comment]
`final int MAX_CARAM = 20;`

Es `CARAMELOS_MAXIMO` o `MAXIMO_DE_CARAMELOS`... Y aun así, en realidad debería ser:

`final int CAPACIDAD_BOLSA = 20;`

**Reply by mmasias** [reply]
Lo mismo, por ejemplo, con las bolsas:

`boolean bolsaUno` o `boolean bolsa1` no dice nada: `boolean bolsaUnoLlena` es otra cosa!!!

**Reply by mmasias** [reply]
```java
if (bolsaDos == false) {
  (...)
}
```

bolsaDos false de qué??? Nótese como palidece cuando se compara con:

```java
if (!bolsaDosLlena) {
  (...)
}
```

o incluso, y mucho mejor:

```java
if (hayEspacioEnBolsaDos) {
  (...)
}
```

**Comment by mmasias** [comment]
Es innecesario multiplicar por 100 para luego "bajarlo" a int (y además dejar un magic number suelto por ahí):

```java
double probabilidadRecibirCaramelos = Math.random() * 100 + 1;
if ((int) probabilidadRecibirCaramelos <= 80) {
  (...)
}
```

Siendo suficiente:

```java
final double PROBABILIDAD_RECIBIR_CARAMELOS = 0.8;

(...)

if (Math.random() < PROBABILIDAD_RECIBIR_CARAMELOS) {
  (...)
}
```

**Reply by mmasias** [reply]
Y de hecho, de esta forma se evita el tener que volver a declarar `probabilidadDeRecibirCaramelos` con cada niño.

**Comment by mmasias** [comment]
Las entidades de un sistema (en nuestro caso, las variables) deben respetar exquisitamente el vocabulario del cliente (en este caso, del enunciado): si el problema habla de edificios, pisos y casas, las variables deben hacer referencia a edificios, pisos y casas; y no a edificaciones, plantas y hogares.

Por tanto, esto:

```java
int i = 1;
while (i <= 4) {
  casasVisitadas++;
  (...)
  i++;
}
```
se reformula a esto:

```java
int casas = 1;
while (casas <= CASAS_POR_PLANTA) {
  casas++;
  casasVisitadas++;
  (...)
}
```

> Lo de casasVisitadas supongo que se puso en previsión a un calculo posterior. No lo termino de entender, pero lo dejo para que se vea como choca en el primer caso y aparentemente sobra en el segundo.

**Comment by mmasias** [comment]
Se empieza desde el principio. Es decir, el primer commit es el código que recibimos. Solo así se puede valorar la magnitud de nuestros ajustes.

**Reply by mmasias** [reply]
Agregando a esto: no tiene ningún sentido que un commit aporte un cambio y su descripción diga otra cosa. Si en un paso se han ajustado nombres de variables, el descriptor del commit debe decirlo. Si además se ha ajustado la indentación, lo mismo: "Indentación y ajuste de nombres": lo que no debemos hacer es decir, por ejemplo, "ajuste de nombres" y que lo que realmente hayamos hecho sea corregir indentación.

**Comment by mmasias** [comment]
Esto

```java
for (int i = 0; (i <= 4) && !(bolsa1 && bolsa2 && bolsa3); i++) 
```

Como vimos en clase, no está mal. Pero, en aras de la claridad, no lo recomendaría.

**Reply by mmasias** [reply]
Y menos aun si no https://github.com/mmasias/24-25-PRG1/discussions/584#discussioncomment-11160395

**Comment by mmasias** [comment]
Los commits deben dejar cargado en el repositorio artefactos en construcción, pero funcionales: si es código, un código que se puede ejecutar, del que previamente hayan validado su funcionamiento.

**Comment by mmasias** [comment]
Si bolsaLlena1 es una variable boolean, entonces no hace falta:

```java
if (bolsaLlena1 == false){
    (...)
}
```

sino que es suficiente con:

```java
if (bolsaLlena1){
    (...)
}
```

**Reply by JavierLianoRioz** [reply]
`bolsaLlena1` es `true` cuando la bolsa esta llena. Por lo que este código se corre cuando bolsaLlena es true. Es decir, cuando la bolsa esté llena:

```
if (bolsaLlena) {
    (...)
}
```

En cambio, en este caso el código se corre cuando bolsaLlena es false. Es decir, cuando la bolsa no esté llena:

```
if (bolsaLlena == false) {
    (...)
}
```

Por lo que lo correcto sería:

```
if (!bolsaLlena) {
    (...)
}
```

**Comment by mmasias** [comment]
Los títulos de los commits (sobretodo cuando estamos corrigiendo código) deben -brevemente- describir lo que ocurre.

**Reply by mmasias** [reply]
Ahh, y el control de versiones está hecho precisamente para que no se tenga que crear un archivo por cada cambio!!!

**Comment by mmasias** [comment]
## Revisión: hoja de seguimiento

- [Parte I](https://docs.google.com/spreadsheets/d/12gwiAT-4JyuX-iZ5CeNSPet8FLALK1SWw2b3F9cxxkk/edit?gid=0#gid=0)
- Parte II

**Comment by mmasias** [comment]
Me han preguntado la diferencia entre el criterio "Variables mal nombradas" y "Nombre pobre de variable". Lo explico:

||||
|-|-|-|
|**Variable mal nombrada**|No sigue el criterio de la comunidad Java para nombrar variables|`final int nombreInadecuado;` en lugar de `final int NOMBRE_ADECUADO;` 
|**Nombre pobre de variable**|El nombre asignado no permite saber el propósito de la variable|https://github.com/mmasias/24-25-PRG1/discussions/584#discussioncomment-11160430

**Comment by mmasias** [comment]
Sé que soy puntilloso con esto, pero creo que es pertinente. Veamos: una cosa es la probabilidad de recibir caramelos, que se expresa en una probabilidad ("hay el 80% de probabilidades de recibir un caramelo") y que por su naturaleza es constante; y otra es la probabilidad contra la que se compara esto en un momento determinado, y dependiendo del caso lo almacenamos o no. 

Con esto en mente, si nos vamos al [ejemplo de Marco](https://github.com/puntoReflex/.github/blob/c782ee4398381e6795f39ac2391c3f2720d6ee73/retos%26proyectos/viajeMarco/src/vPRG1/Marco.java#L27) vemos que por un lado se define una constante `HARD_RAIN` (que ya puestos bien podría ser `HARD_RAIN_PROBABILITY`) que almacena esta probabilidad, y por otro lo que ocurre en el momento en la forma de variable: `rainProbability`, que es la que recibe el Math.random(). Esta se almacena solo porque tendremos luego que volverla a utilizar. Por ejemplo, en el caso del mono [no se almacena y se compara al vuelo](https://github.com/puntoReflex/.github/blob/c782ee4398381e6795f39ac2391c3f2720d6ee73/retos%26proyectos/viajeMarco/src/vPRG1/Marco.java#L36).

**Comment by mmasias** [comment]
En lugar de:

```java
if (bolsaLlena1 && bolsaLlena2 && bolsaLlena3) {
    todasLasBolsasLlenas = true;
}
```

es mejor usar:

```java
todasLasBolsasLlenas = bolsaLlena1 && bolsaLlena2 && bolsaLlena3;
```
